esphome/components/modbus_controller/modbus_controller.cpp

```diff
@@ -277,12 +280,13 @@ void ModbusController::dump_config() {
 #if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_VERBOSE
   ESP_LOGCONFIG(TAG, "sensormap");
   for (auto &it : sensorset_) {
-    ESP_LOGCONFIG(TAG, "  Sensor start=0x%X count=%d size=%d", it->start_address, it->register_count,
+    ESP_LOGCONFIG(TAG, " Type=%zu Sensor start address=0x%X, offset=0x%X count=%d size=%d",
```

Comment:
Not sure, but I would prefer `Sensor type=%zu start address=0x%x...` just to be similar to the log message of the range at line 289.